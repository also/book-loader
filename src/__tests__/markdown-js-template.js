function createMd() {
  const markdownJsTemplate = require('../markdown-js-template');
  const markdownIt = require('markdown-it');
  return markdownIt().use(markdownJsTemplate);
}

test('handles links', () => {
  const md = createMd();
  const result = md.render('[a test](test.md)');
  expect(result).toMatchSnapshot();
});

test('allows templates in code blocks', () => {
  const md = createMd();
  const result = md.render('```\n<%= require("content") %>\n```');
  expect(result).toMatchSnapshot();
});

xtest('allows templates in links', () => {
  const md = createMd();
  const result = md.render(`[link](<%= require('target') %>)`);
  expect(result).toMatchSnapshot();
});
