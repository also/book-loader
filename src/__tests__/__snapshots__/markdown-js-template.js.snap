exports[`test allows templates in code blocks 1`] = `
"\"<pre><code>\" +
    ( require(\"content\") ) +
    \"\\n</code></pre>\\n\""
`;

exports[`test handles links 1`] = `
"\"<p><a href=\\\"\" +
    (book.pageUrl(\"./test.md\")) +
    \"\\\">a test</a></p>\\n\""
`;
